### ISSUE TITLE
Rendering mode on android

### ISSUE THREAD

**Issue by NewFURRY3DSXL** [JordanSantiagoYT/FNF-JS-Engine#672]
### What feature do you want to get added on the **base** engine?

I would like to add rendering mode to android, or at least an optimized mode to record my screen and showcase spammy charts with no lag.

I'm not entirely sure, but somehow we could use an ffmpeg build and maybe create a folder called "renders" on the game folder and save the rendered videos there!

If I'm not entirely wrong there might be already ffmpeg encoding apps! Maybe install one of those and render them charts there!

this is my first time posting a feature request, if I did something wrong tell me! :3

### To test your sight, and reliability, please select the option of what should NOT be requested.

Proper credit that was forgotten

**Comment by JordanSantiagoYT** [comment]
failed the test lol

i will consider adding this if there does surface undeniable proof that a working ffmpeg (that can encode image sequences) exists, both on Android AND iOS.

**Comment by PatoFlamejanteTV** [comment]
> failed the test lol
> 
> i will consider adding this if there does surface undeniable proof that a working ffmpeg (that can encode image sequences) exists, both on Android AND iOS.

https://play.google.com/store/apps/details?id=com.crossplat.ffmpegmobile

**Comment by PatoFlamejanteTV** [comment]
https://github.com/tanersener/mobile-ffmpeg (-> https://github.com/tanersener/ffmpeg-kit)
https://www.geeksforgeeks.org/how-to-use-ffmpeg-in-android-with-example/
https://apkpure.com/ffmpeg-media-encoder/com.silentlexx.ffmpeggui/

**Comment by PatoFlamejanteTV** [comment]
[https://github.com/arthenica/ffmpeg-kit](https://github.com/arthenica/ffmpeg-kit)

[https://stackoverflow.com/questions/11329346/how-to-build-and-link-ffmpeg-to-ios](https://stackoverflow.com/questions/11329346/how-to-build-and-link-ffmpeg-to-ios)

https://codingtube.tech/2021/12/03/ffmpeg/how-to-use-ffmpeg-in-ios/

**Comment by PatoFlamejanteTV** [comment]
Btw FFMpeg-kit is very complete, supports basically all devices/OSes:

![](https://github.com/user-attachments/assets/1c23b873-31ed-410a-bfa7-9a6768a111df)

**Comment by JordanSantiagoYT** [comment]
Technically added to mobile branch in most recent commit.

**Comment by PatoFlamejanteTV** [comment]
YAYYY!!! :D

**Comment by mcagabe19** [comment]
I reverted the commit cuz we have no access to that ffmpeg

**Comment by mcagabe19** [comment]
we need ffmpeg binaries for native android and iOS in the js engine which is already impossible

**Comment by moxie-coder** [comment]
> we need ffmpeg binaries for native android and iOS in the js engine which is already impossible

got it hun

**Comment by PatoFlamejanteTV** [comment]
> we need ffmpeg binaries for native android and iOS in the js engine which is already impossible

You mean the .so/DLLs?

**Comment by moxie-coder** [comment]
It’s more complicated than that

**Comment by mcagabe19** [comment]
I found ffmpreg adrod binary (that's very old) maybe that can be done
![image](https://github.com/user-attachments/assets/6e68b1f0-bf46-44ea-95ce-c18802b1bc91)

**Comment by mcagabe19** [comment]
https://github.com/JordanSantiagoYT/FNF-JS-Engine/commit/dc858d66fbe01eaabfcb32aa52793be860be43e4

**Comment by JordanSantiagoYT** [comment]
wait

am i able to release 1.37.0 no

**Comment by mcagabe19** [comment]
up to you jordan

**Comment by mcagabe19** [comment]
@JordanSantiagoYT but know that 1.37.0 will have the https://github.com/JordanSantiagoYT/FNF-JS-Engine/commit/dc858d66fbe01eaabfcb32aa52793be860be43e4 commit

**Comment by mcagabe19** [comment]
alright seems like `sys.io.Process` doesn't want to work and that's probably haxe's issue I don't know, but this issue is keeping opened cuz there's same issue on `extension-androidtools` that isRooted also hangs up the game where it calls `su`.

**Comment by xdczmods** [comment]
Fps​ 30-120
Fps​ 1000=🎇

**Comment by StinkTheStinker** [comment]
guessing it's technically complete because you can just get a screen recorder

**Comment by moxie-coder** [comment]
> guessing it's technically complete because you can just get a screen recorder

pretty much, you don't need ffmpeg then

**Comment by mcagabe19** [comment]
> guessing it's technically complete because you can just get a screen recorder

not quite, ffmpeg is implemented but `sys.io.Process` still causing issues

**Comment by mcagabe19** [comment]
https://youtu.be/4_vbx9SKTO8

**Comment by mcagabe19** [comment]
I need build ffmpeg ugh

**Comment by mcagabe19** [comment]
https://www.youtube.com/channel/UC2cw3Qm8r9Luxjxxk9BO0Wg/community?lb=UgkxJb6Uxc0YAnxUoi04EPRt-3S9uEWsGxye

**Comment by SyncGit12** [comment]
updates?

**Comment by mcagabe19** [comment]
okay so this ffmpeg only works on AOSP, but CRM works fine (need termux to use ffmpeg manually) when I got time I'll remove ffmpeg and edit render finish state

**Comment by mcagabe19** [comment]
couldn't figure out non-CRM so this will be CRM only with the current commit on JS Engine Mobile.
